File: week2-Arrays/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int Coleman_Liau_alg(int letters, int sentences, int words);

int letterCounter(string text);

int wordCounter(string text);

int sentenceCounter(string text);

int main(void)
{
    // Getting the text
    string text = get_string("Text: ");

    int numLetters = letterCounter(text);
    int numWords = wordCounter(text);
    int numSentences = sentenceCounter(text);

    // printf("letters: %d\n", numLetters);
    // printf("words: %d\n", numWords);
    // printf("senteces: %d\n", numSentences);

    int grade = Coleman_Liau_alg(numLetters, numSentences, numWords);

    // If it is before grade 1
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }

    // If it is 16 or above
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %d\n", grade);
    }

    return 0;
}

int Coleman_Liau_alg(int letters, int sentences, int words)
{
    // Implemented based on the Coleman-Liau algorithm to determine the reading level of a text
    float L = ((float) letters / words * 100);
    float S = ((float) sentences / words * 100);

    float grade = 0.0588 * L - 0.296 * S - 15.8;

    return round(grade);
}

int letterCounter(string text)
{
    int counter = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        // Checking if the char is alphabetical
        if (isalpha(text[i]))
        {
            counter++;
        }
    }

    return counter;
}

int wordCounter(string text)
{
    int counter = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        // Checking if the char ascii value == 32, which is space
        if ((int) text[i] == 32)
        {
            counter++;
        }
    }

    // Number of words in a text == number of spaces + 1
    return counter + 1;
}

int sentenceCounter(string text)
{
    int counter = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        // Checking if the char is . or ! or ?
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            counter++;
        }
    }

    return counter;
}

File: week6-Python/sentimental-readability/readability.py
from cs50 import get_string


def main():
    # Getting the text from the user
    text = get_string("Text: ")

    numLetters = letterCounter(text)
    numWords = wordCounter(text)
    numSentences = sentenceCounter(text)

    grade = Coleman_Liau_alg(numLetters, numWords, numSentences)

    # If then else statements to return the grade to the user
    if grade < 1:
        print("Before Grade 1")

    elif grade >= 16:
        print("Grade 16+")

    else:
        print(f"Grade {grade}")


def letterCounter(text_input):
    counter = 0

    for char in text_input:
        # Checking if char is alphabetical, if so, it's a letter
        if char.isalpha():
            counter += 1

    return counter


def wordCounter(text_input):
    counter = 0

    for char in text_input:
        # Checking for spaces in text
        if ord(char) == 32:
            counter += 1

    # Number of words in a text = num of spaces + 1
    return counter + 1


def sentenceCounter(text_input):
    counter = 0

    for char in text_input:
        # Checking if it is a char that ends a sentence
        if char == '.' or char == '!' or char == '?':
            counter += 1

    return counter


def Coleman_Liau_alg(letters, words, sentences):
    # Implemented based on the Coleman-Liau algorithm to determine the reading level of a text
    L = letters / words * 100
    S = sentences / words * 100

    grade = 0.0588 * L - 0.296 * S - 15.8

    return round(grade)


if __name__ == "__main__":
    main()